From 47c490935e12503424de5c20e567466f741fd69f Mon Sep 17 00:00:00 2001
From: AsavarTzeth <asavartzeth@gmail.com>
Date: Mon, 23 Sep 2019 11:35:44 +0200
Subject: [PATCH] Add configure option to disable documentation

Effectively this makes asciidoc/a2x an optional dependency, which is
particularly useful if python2 is unavailable.

This will be an increasingly common issue since python2 is near
end-of-life and a2x is yet to make a python3 compatible release.
---
 Makefile.am  | 4 ++++
 configure.ac | 8 +++++++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index cc2ef4b..bb47f4e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -63,6 +63,7 @@ ykparse_LDADD = ./libyubikey.la
 ykgenerate_SOURCES = ykgenerate.c
 ykgenerate_LDADD = ./libyubikey.la
 
+if HAVE_A2X
 dist_man_MANS = modhex.1 ykparse.1 ykgenerate.1
 DISTCLEANFILES = $(dist_man_MANS)
 
@@ -72,6 +73,9 @@ SUFFIXES = .1.txt .1
 
 MANSOURCES = modhex.1.txt ykparse.1.txt ykgenerate.1.txt
 EXTRA_DIST = $(MANSOURCES) simple.mk THANKS
+else
+EXTRA_DIST = simple.mk THANKS
+endif
 
 # Maintainer rules
 
diff --git a/configure.ac b/configure.ac
index f409716..8c600b2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -48,7 +48,13 @@ AM_PROG_AR
 
 AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
-AM_MISSING_PROG([A2X], a2x, $missing_dir)
+
+AC_ARG_ENABLE([documentation],
+  [AS_HELP_STRING([--disable-documentation],
+		  [turn off building of man pages @<:@default=auto@:>@])])
+AS_IF([test "x$enable_documentation" != xno],
+  [AM_MISSING_PROG([A2X], a2x, $missing_dir)])
+AM_CONDITIONAL(HAVE_A2X, [test "x$enable_documentation" != xno])
 
 gl_LD_VERSION_SCRIPT
 gl_VALGRIND_TESTS
-- 
2.23.0

